### DISCUSSION TITLE
useFetch key best practices

### DISCUSSION THREAD

**Discussion by danielkellyio** [nuxt/nuxt#16852] [Questions]
When using the key option for useFetch what are the best practices around generating that key? This should be unique across the entire application correct? Would the page route suffice in the case of a dynamic id? I've seen some hashing the key. Is this recommended/helpful? https://github.com/nuxt/framework/issues/5993#issuecomment-1191095162

```
const { data } = await useFetch(
  `http://www.omdbapi.com/?i=${useRoute().params.id}`,
  {
    key: `movies/id/${useRoute().params.id}` // what's the best thing here?
  }
);
```

**Comment by kentntwari** [comment]
This is actually a valid question. I ran into a case where I wanted to cache the data depending on the key so I thought of making it dynamic but it didn't seem to work. So yeah it should be clear what exactly are best practices around using a key for useFetch

**Comment by ndragun92** [comment]
Well it depends. If you want to share it you can leave something simple.

If you are dealing with pagination/infinite load and such you can extend that key to be something more dynamic.

Also if content on every page is different and you dont want to share it you can change it.

For example if you store data with key `mytestkey`

You can then fetch it anywhere like `const data = useNuxtData('mytestkey')`

That way you can easy share data since things stored in nuxt payload are globally available and acts similar to pinia store

You can also use on useFetch level


```
getCachedData: (key, nuxtApp) => {
    console.log('getCachedData', key)
    // Return from cache if exists and prevent additional load
    return nuxtApp.payload.data[key] || nuxtApp.static.data[key]
  }
```

In this case if it would find cached data it would skip making a request on server. When this check returns undefined then its making a new call to API and will store it in payload.

Hope it helps!
@danielkellyio
